src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java

```diff
@@ -17,20 +18,24 @@
     private static final String SUCCESS = "Success";
     private static final String FAIL = "Fail";    
 
+    private final Pattern diceMessageRgx;
     private final Pattern diceFaceRgx;
     private final Pattern modRgx;
 
     private final CommandHandler commandHandler;
 
     public RedMarketCommand(CommandHandler commandHandler) {
         this.commandHandler = commandHandler;
-        diceFaceRgx = Pattern.compile("(black|b|red|r)(\\d+)");
+        diceMessageRgx = Pattern.compile("((black|b|red|r)\\d+|\\d+(black|b|red|r))");
```

Comment:
Running regex twice for almost the exact same pattern seems wasteful. But you're also fixing this bug before I am so I'll probably allow it and eventually make a "performance enhancement" to get this back down to one regex.

Reply:
Yes im aware of that. I read the matcher could handle nested capture groups, but unfortunately I couldn't make it work. I had trouble with the two possibilities canceling eachother out